archunit/src/test/java/com/tngtech/archunit/core/domain/JavaPackageTest.java

```diff
@@ -241,77 +242,156 @@ public void visits_packages() {
+    public void has_package_dependencies_from_this_package() {
+        JavaPackage examplePackage = importPackage("packageexamples");
+
+        assertThat(examplePackage.getPackage("second").getPackageDependenciesFromThisPackage())
+                .containsOnly(
+                        getRoot(examplePackage).getPackage("java.lang"),
+                        examplePackage.getPackage("second.sub"),
+                        examplePackage.getPackage("first"));
 
-        assertThatDependencies(examplePackage.getPackage("unrelated").getClassDependenciesToSelf())
+        assertThat(examplePackage.getPackage("third").getPackageDependenciesFromThisPackage())
                 .isEmpty();
+
+        assertThatPackages(examplePackage.getPackage("unrelated").getPackageDependenciesFromThisPackage())
+                .containOnlyNames("java.lang");
     }
 
     @Test
-    public void has_package_dependencies_to_other_packages() {
+    public void has_package_dependencies_from_this_package_and_subpackages() {
         JavaPackage examplePackage = importPackage("packageexamples");
 
-        assertThat(examplePackage.getPackage("second").getPackageDependenciesFromSelf())
+        assertThat(examplePackage.getPackage("second").getPackageDependenciesFromThisPackageTree())
                 .containsOnly(
                         getRoot(examplePackage).getPackage("java.lang"),
                         examplePackage.getPackage("first"),
                         examplePackage.getPackage("third.sub"));
 
-        assertThat(examplePackage.getPackage("third").getPackageDependenciesFromSelf())
+        assertThat(examplePackage.getPackage("third").getPackageDependenciesFromThisPackageTree())
                 .containsOnly(examplePackage.getPackage("first"));
 
-        assertThatPackages(examplePackage.getPackage("unrelated").getPackageDependenciesFromSelf())
+        assertThatPackages(examplePackage.getPackage("unrelated").getPackageDependenciesFromThisPackageTree())
                 .containOnlyNames("java.lang");
     }
 
     @Test
-    public void has_package_dependencies_from_other_packages() {
+    public void has_package_dependencies_to_this_package() {
         JavaPackage examplePackage = importPackage("packageexamples");
 
-        assertThat(examplePackage.getPackage("first").getPackageDependenciesToSelf())
+        assertThat(examplePackage.getPackage("first").getPackageDependenciesToThisPackage())
                 .containsOnly(
                         examplePackage.getPackage("second"),
                         examplePackage.getPackage("second.sub"),
                         examplePackage.getPackage("third.sub"));
 
-        assertThat(examplePackage.getPackage("third").getPackageDependenciesToSelf())
-                .containsOnly(examplePackage.getPackage("second.sub"));
+        assertThat(examplePackage.getPackage("second").getPackageDependenciesToThisPackage())
+                .doesNotContain(examplePackage.getPackage("second"));
```

Comment:
```suggestion
        assertThat(examplePackage.getPackage("second").getPackageDependenciesToThisPackage())
                .containsOnly(
                        examplePackage.getPackage("first"),
                        examplePackage.getPackage("second.sub"));
```
This would probably be a stronger assertion